Data/models/Bookmark.swift

```diff
@@ -170,12 +173,28 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
                    save: Bool = false,
                    sendToSync: Bool = false,
                    parentFolder: Bookmark? = nil,
-                   context: NSManagedObjectContext = DataController.newBackgroundContext()) -> Bookmark? {
+                   context: NSManagedObjectContext? = nil) {
+        
+        if let context = context {
```

Comment:
To give another example from convo above:

```
let performingContext = context ?? DataController.newBackgroundContext
DataController.perform(with: performingContext) { // See convo on `performBackgroundTask` to understand this
    addImpl(rootObject: root, save: save, sendToSync: sendToSync, parentFolder: parentFolder, context: backgroundContext)
}
```

Or could even just throw a context at `addImpl` and have it deal with this internally.

Reply:
Sounds good to me, I wasn't exactly sure how to simplify all these `if let context = context {..` code I added